### ISSUE TITLE
read.odf() error if a unit is missing

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1152]
This may relate to some other recent ODF issues and code changes. I plan to have a look at this today. Possibly my fix will conflict with things in Clark's `cr` branch, but we can always sort such things out later, I think. (PS. the file is not public.)

```R
/data/flemishCap/moorings/1840/ProcessedMC/MCM_HUD2013021_1840_2305_300.ODF 
Error in if ("" != as.character(namesUnits$units[[w]]$unit)) warning("ODF column \"",  : 
  argument is of length zero
Calls: read.oce -> read.odf
```

**Comment by dankelley** [comment]
The metadata in this file indicate that something special was done with the `CRAT` column. Perhaps for that reason, its information block lacks a `UNITS` designation. That's the cause of the problem.

**Comment by dankelley** [comment]
Fixed in `develop` commit 808de8a374285a4eec989394287c7554f0f4bd47

Part of it was that `as.character(expression())` does not yield `""` as I had been thinking, but rather it returns `character(0)`. So I fixed that, and grepped around to see if any similar problems might be hidden (none found, but this note will be a reminder to check).
